sdk/trace/id_generator.go

```diff
@@ -15,23 +15,29 @@
 package trace // import "go.opentelemetry.io/otel/sdk/trace"
 
 import (
+	crand "crypto/rand"
+	"encoding/binary"
 	"math/rand"
 	"sync"
 
 	"go.opentelemetry.io/otel/trace"
-
-	"go.opentelemetry.io/otel/sdk/trace/internal"
 )
 
-type defaultIDGenerator struct {
+// IDGenerator allows custom generators for TraceID and SpanID.
+type IDGenerator interface {
+	NewTraceID() trace.TraceID
+	NewSpanID() trace.SpanID
```

Comment:
I like @jmacd's idea of having these be a function that returns a span ID and a trace ID with a span ID. I also wonder if we can include arguments that would be helpful for ID generation. Maybe something like:

```suggestion
	NewIDs(ctx context.Context) (trace.TraceID, trace.SpanID)
	NewSpanID(ctx context.Context, traceID trace.TraceID) trace.SpanID
```

I think passing a context will enable generation with the context of things like baggage or other vendor defined values. Also, passing the trace ID to the NewSpanID method might be a useful things if a generator uses that in relation to the span IDs it generates.